src/main/java/net/neoforged/neoforge/server/jsonrpc/NeoForgeSchemas.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.server.jsonrpc;
+
+import net.minecraft.server.jsonrpc.api.ReferenceUtil;
+import net.minecraft.server.jsonrpc.api.Schema;
+import net.minecraft.server.jsonrpc.api.SchemaComponent;
+
+public final class NeoForgeSchemas {
+    private NeoForgeSchemas() {}
+
+    public static final SchemaComponent MOD_SCHEMA = registerSchema("mod", Schema.record()
+            .withField("modId", Schema.STRING_SCHEMA.flatten())
+            .withField("version", Schema.STRING_SCHEMA.flatten())
+            .withField("displayName", Schema.STRING_SCHEMA.flatten())
+            .withField("description", Schema.STRING_SCHEMA.flatten()));
+
+    private static SchemaComponent registerSchema(String name, Schema schema) {
+        SchemaComponent schemacomponent = new SchemaComponent("neoforge:" + name, ReferenceUtil.createLocalReference(name), schema);
+        Schema.getSchemaRegistry().add(schemacomponent);
```

Comment:
This list is not thread safe, registering the schema would also need to be done through an event.

Reply:
You don't need the schema creation to be deferred? Just the registration